dracut-saltboot/saltboot/saltboot-root.sh

```diff
@@ -6,6 +6,7 @@ else
   # we have root, we can eventually continue without network after timeout
   # the timeout is controlled by rd.retry option
   echo "rm -f -- $hookdir/initqueue/finished/wait-network.sh" > $hookdir/initqueue/timeout/saltboot-network.sh
+  echo "rm -f -- $hookdir/initqueue/finished/devexists-*" > $hookdir/initqueue/timeout/saltboot-device.sh
 fi
```

Comment:
If both network and root device is missing, it should end in emergency shell after timeout. I am not sure if it works with this. I think we should delete just one "finished" check.

Reply:
Finally I managed to reproduce this scenario.
For future reference what I needed to do:
1) create saltboot partitioning with bios_grub partition as documented, but also create separate /boot partition
2) this separate /boot partition is however mounted only after initrd run, so after boot this directory is empty. Thus I needed to call mkinitrd manually
3) disable dhcp on branch server, remove system partition, reboot system

After that terminal boots locally from the boot partition, waits 240s then timeout handles are removed and boot continue. pre-mount step pass without any message because network is unavailable saltboot is skipped.
Then it continues enabling systemd services when it starts waiting for the missing device with unlimited time.

However I should note that it required manual steps to get to this state. I think that the only one other time how to get to this state is that after harddisk replacement network works for the pxe part, but then is cut short before dracut is started. However result will be in my opinion the same either from emergency shell or stuck in systemd service waiting forever -> restart of the terminal.

Reply:
I updated the PR with selective removal of the waiters. Now if either network or disk is available it will remove both waiters. But if both are still there and waiting it will wait and timeout to the dracut emergency shell. WDYT?